google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineFacetConfiguration.java

```diff
@@ -69,19 +71,11 @@ public void writeExternal(Element element) throws WriteExternalException {
    * returned and therefore not considered to be "managed".
    */
   public Set<AppEngineStandardMavenLibrary> getLibraries(@NotNull Project project) {
-    LibraryTable libraryTable = LibraryTablesRegistrar.getInstance().getLibraryTable(project);
-    Library[] libraries =  libraryTable.getLibraries();
-
-    Set<AppEngineStandardMavenLibrary> configuredLibraries = Sets.newHashSet();
-    for (Library configuredLibrary : libraries) {
-      AppEngineStandardMavenLibrary mavenLibrary
-          = AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(configuredLibrary.getName());
-      if (mavenLibrary != null) {
-        configuredLibraries.add(mavenLibrary);
-      }
-    }
-
-    return configuredLibraries;
+    return Arrays.stream(
+        LibraryTablesRegistrar.getInstance().getLibraryTable(project).getLibraries())
+        .map(library ->
+            AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(library.getName()).get())
```

Comment:
> does this still do that?

I don't think it will. `map()` in any functional language never removes an element, i.e., the resulting collection has the same size.